lld/ELF/Arch/Mips.cpp

```diff
@@ -83,6 +84,16 @@ template <class ELFT> uint32_t MIPS<ELFT>::calcEFlags() const {
   return calcMipsEFlags<ELFT>();
 }
 
+template <class ELFT> bool MIPS<ELFT>::calcIsCheriAbi() const {
+  bool isCheriAbi = (config->eflags & EF_MIPS_ABI) == EF_MIPS_ABI_CHERIABI;
```

Comment:
Unrelated to this but it looks like calcMipsEFlags() returns `config->mipsN32Abi ? EF_MIPS_ABI2 : EF_MIPS_ABI_O32` for binary files instead of zero when a -m flag is passed.
It should probably also look at config->isCheriAbi or always return zero.

Reply:
Yeah we can probably remove it on MIPS. I think it was just there because we had a custom _fbsd_cheri.x default linker script in BFD and I had to support both while LLD was experimental.

Reply:
Yay. We can do that as a separate change though; this PR allows CheriBSD to migrate without needing a synchronised flag day.